### PR TITLE
#52 Add LogRocket and #53 improve error message

### PR DESCRIPTION
**Pull request by VincentBean**
Added LogRocket issue #52 
Improved error message issue #53

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -125,11 +126,37 @@ public function collectModuleConfig()
...
+            $reason .= __('Module is not enabled in config. ');
+        }
+        if (!$dsnNotEmpty) {
+            $reason .= __('DSN is empty. ');
+        }
+        if (!$productionMode) {
+            $reason .= __('Not in production and development mode is false.');
+        }
+
+        return !$emptyConfig && $configEnabled && $dsnNotEmpty && $productionMode;
```
Comment (indykoning) [inline Helper/Data.php:159]:
```suggestion
    public function isActiveWithReason()
    {
        $reasons = [];
        $emptyConfig = empty($this->config);
        $configEnabled = array_key_exists('enabled', $this->config) && $this->config['enabled'];
        $dsnNotEmpty = $this->getDSN();
        $productionMode = ($this->isProductionMode() || $this->isOverwriteProductionMode());

        if ($emptyConfig) {
            $reasons[] = __('Config is empty. ');
        }
        if (!$configEnabled) {
            $reasons[] = __('Module is not enabled in config. ');
        }
        if (!$dsnNotEmpty) {
            $reasons[] = __('DSN is empty. ');
        }
        if (!$productionMode) {
            $reasons[] = __('Not in production and development mode is false.');
        }

        return strlen($reason) ? ['active' => 'false', 'reasons' => $reason] : ['active' => true];
```

File: Helper/Data.php [hunk 2]
```diff
@@ -125,11 +126,37 @@ public function collectModuleConfig()
...
     public function isActive()
     {
-        return !empty($this->config)
-            && array_key_exists('enabled', $this->config)
-            && $this->config['enabled']
-            && $this->getDSN()
-            && ($this->isProductionMode() || $this->isOverwriteProductionMode());
+        $dummyStr = '';
+
+        return $this->isActiveWithReason($dummyStr);
```
Comment (indykoning) [inline Helper/Data.php:131]:
```suggestion
        return $this->isActiveWithReason()['active'];
```

File: Controller/Adminhtml/Test/Sentry.php [hunk 3]
```diff
@@ -84,7 +84,9 @@ public function execute()
     {
         $result = ['status' => false];
 
-        if ($this->helperSentry->isActive()) {
+        $inactiveReason = '';
+
+        if ($this->helperSentry->isActiveWithReason($inactiveReason)) {
```
Comment (indykoning) [inline Controller/Adminhtml/Test/Sentry.php:89]:
```suggestion
        if (($activeWithReason = $this->helperSentry->isActiveWithReason()) && $activeWithReason['active']) {
```

File: Controller/Adminhtml/Test/Sentry.php [hunk 4]
```diff
@@ -94,7 +96,7 @@ public function execute()
                 $this->logger->critical($e);
             }
         } else {
-            $result['content'] = __('Sentry Domain not filled!');
+            $result['content'] = $inactiveReason;
```
Comment (indykoning) [inline Controller/Adminhtml/Test/Sentry.php:99]:
```suggestion
            $result['content'] = implode(PHP_EOL, $activeWithReason['reason']);
```

File: view/frontend/layout/default.xml [hunk 5]
```diff
@@ -2,11 +2,11 @@
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <body>
         <referenceBlock name="head.additional">
-            <block class="JustBetter\Sentry\Block\SentryScript" before="-" template="JustBetter_Sentry::script/sentry.phtml" name="sentry.head.additional" />
+            <block class="JustBetter\Sentry\Block\SentryScript" before="-" template="JustBetter_Sentry::script/sentry.phtml" name="sentry.head.additional" cacheable="false"  />
```
Comment (indykoning) [inline view/frontend/layout/default.xml:5]: Using cacheable="false" disables Full page caching for the whole page
https://magento.stackexchange.com/a/103319

Maybe load the customer module with requirejs to get the customer info
https://magento.stackexchange.com/questions/134466/how-to-check-that-customer-is-logged-in-on-frontend-in-magento-2-1

Comment (VincentBean) [reply view/frontend/layout/default.xml]: See commit https://github.com/justbetter/magento2-sentry/pull/55/commits/31cf934454b2d6ea6c20706eee8a6a71e750c6eb

**Pull request merged**